python/sglang/srt/managers/controller/tp_worker.py

```diff
@@ -711,8 +768,20 @@ def handle_finished_requests(self, batch: Batch):
             else:
                 batch.reqs = []
 
+    def filter_out_inflight(self, batch: Batch):
+        if self.current_inflight_req is None:
+            return
+
+        unfinished_indices = list(range(len(batch.reqs)))
+        for i, req in enumerate(batch.reqs):
+            if req is self.current_inflight_req:
+                unfinished_indices.remove(i)
+                break
+
+        batch.filter_batch(unfinished_indices)
```

Comment:
Here, we only drop one index. Maybe we can have a specialized version for this.

Reply:
I have no idea how to simplify it yet. I will try in the next PR.